### PR TITLE
old repo commit

### PR DESCRIPTION
**Pull request by paddyduff31**

### CODE REVIEW

File: src/ParticipantApi/ParticipantApi.csproj [hunk 1]
```diff
@@ -24,8 +24,8 @@
     <PackageReference Include="Amazon.Lambda.Serialization.Json" Version="2.0.0" />
     <PackageReference Include="AWSSDK.Core" Version="3.7.6.4" />
     <PackageReference Include="AWSSDK.SecretsManager" Version="3.7.2.2" />
-    <PackageReference Include="Dte.Common" Version="1.4.0" />
-    <PackageReference Include="Dte.Common.Lambda" Version="1.1.3" />
+    <PackageReference Include="Dte.Common" Version="1.6.0" />
+    <PackageReference Include="Dte.Common.Lambda" Version="0.0.4" />
```
Comment (con-nihr) [inline src/ParticipantApi/ParticipantApi.csproj:28]: Don't downgrade

File: tests/ParticipantApi.Acceptance.Tests/ParticipantApi.Acceptance.Tests.csproj [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-        <TargetFramework>netcoreapp3.1</TargetFramework>
+        <TargetFramework>net6.0</TargetFramework>
 
         <IsPackable>false</IsPackable>
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="Dte.Api.Acceptance.Test.Helpers" Version="1.0.2" />
-        <PackageReference Include="Dte.Common" Version="1.4.0" />
+        <PackageReference Include="Dte.Api.Acceptance.Test.Helpers" Version="0.0.5" />
```
Comment (con-nihr) [inline tests/ParticipantApi.Acceptance.Tests/ParticipantApi.Acceptance.Tests.csproj:10]: Don't downgrade

**Review by con-nihr — CHANGES_REQUESTED** [review]

File: src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs [hunk 3]
```diff
@@ -41,15 +41,11 @@ public async Task<ParticipantDetails> GetParticipantDetailsAsync(string particip
                 _config);
         }
 
-        public async Task<ParticipantDetails> GetParticipantDetailsByEmailAsync(string email)
+        private async Task<ParticipantDetails> GetParticipantDetailsByFilterAsync(
+            string dbCol,
+            string filterKey,
```
Comment (con-nihr) [inline src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs:46]: Remove `filterKey`

File: src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs [hunk 4]
```diff
@@ -59,16 +55,16 @@ public async Task<ParticipantDetails> GetParticipantDetailsByEmailAsync(string e
                 var request = new ScanRequest
                 {
                     TableName = _config.OverrideTableName,
-                    FilterExpression = "#email = :email",
+                    FilterExpression = $"{filterKey} = :{filterKey}",
```
Comment (con-nihr) [inline src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs:58]: `FilterExpression = $"#{dbCol} = :{dbCol}",`

File: src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs [hunk 5]
```diff
@@ -59,16 +55,16 @@ public async Task<ParticipantDetails> GetParticipantDetailsByEmailAsync(string e
...
                     ExpressionAttributeNames = new Dictionary<string, string>
                     {
                         {
-                            "#email", "Email"
+                            $"#{filterKey}", dbCol
```
Comment (con-nihr) [inline src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs:62]: `$"#{dbCol}", dbCol`

File: src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs [hunk 6]
```diff
@@ -59,16 +55,16 @@ public async Task<ParticipantDetails> GetParticipantDetailsByEmailAsync(string e
...
                         }
                     },
                     ExpressionAttributeValues = new Dictionary<string, AttributeValue>
                     {
-                        { ":email", new AttributeValue { S = email } }
+                        { $":{filterKey}", filterValue }
```
Comment (con-nihr) [inline src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs:67]: `{ $":{dbCol}", filterValue }`

File: src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs [hunk 7]
```diff
@@ -41,15 +41,11 @@ public async Task<ParticipantDetails> GetParticipantDetailsAsync(string particip
                 _config);
         }
 
-        public async Task<ParticipantDetails> GetParticipantDetailsByEmailAsync(string email)
+        private async Task<ParticipantDetails> GetParticipantDetailsByFilterAsync(
```
Comment (con-nihr) [inline src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs:44]: Make it explicit that this is a scan operation.
`ScanForParticipantDetailsWithFilterAsync`

**Review by con-nihr — CHANGES_REQUESTED** [review]

File: src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs (hunk 4 above)
Comment (paddyduff31) [reply src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs:58]: So dbCol is capitalised and filter key is not, is it better to do this through code?

File: src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs [hunk 8]
```diff
@@ -55,16 +54,16 @@ public async Task<ParticipantDetails> GetParticipantDetailsAsync(string particip
                 var request = new ScanRequest
                 {
                     TableName = _config.OverrideTableName,
-                    FilterExpression = $"{filterKey} = :{filterKey}",
+                    FilterExpression = $"{dbCol} = :{dbCol}",
```
Comment (con-nihr) [inline src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs:57]: Should be `FilterExpression = $"#{dbCol} = :{dbCol}",`

**Review by con-nihr — CHANGES_REQUESTED** [review]
Small tweak to FilterExpression

**Review by con-nihr — APPROVED** [review]

**Pull request merged**